contrib/codegen/tests/ui-fail/database-types.rs

```diff
@@ -7,6 +7,8 @@ struct Unknown;
 //~^ ERROR Unknown: rocket_contrib::databases::Poolable
 //~^^ ERROR is private
 //~^^^ ERROR no method named `get`
+//~^^^^ ERROR no function or associated item
+//~^^^^^ ERROR Unknown: rocket_contrib::databases::Poolable
```

Comment:
They might be indirectly caused by these changes. I have noticed that in general rustc has been more and more likely to continue with further error checking over the last few months, so it is hard to say exactly.

Reply:
As far as I can tell, the `no function or associated item` is due to attempting to call `.map(#conn_type::pool)` - incidentally, that should probably be `<#conn_type as #Poolable>::pool`.

This playground does some of the same things that `#[database]` does when the types don't implement `Poolable`:

https://play.rust-lang.org/?version=nightly&mode=debug&edition=2018&gist=6c3e950eafb9094cf567495fb5d0cddd

Notice that invoking a nonexistent method on `b` in an async block is enough to raise an error and also suppress the errors indicating that `A` and `B` are not iterators, but calling it directly in a function doesn't do the same (seemingly because `S` itself is determined to be malformed before even typechecking anything in `impl S`). It's likely that the number of errors will continue to go up and down over time with minor tweaks to resolution order of types and traits as it has in the past.